lib/api/WP_Auth0_Api_Abstract.php

```diff
@@ -180,47 +182,37 @@ protected function set_path( $path ) {
 	 */
 	protected function set_bearer( $scope ) {
 
-		$this->api_token = wp_cache_get( self::CACHE_KEY, WPA0_CACHE_GROUP );
-		if ( ! $this->api_token ) {
-			$this->api_token = $this->options->get( 'auth0_app_token' ); // REMOVE
-		}
+		$api_token = WP_Auth0_Api_Client_Credentials::get_stored_token();
 
-		if ( $this->api_token ) {
-			try {
-				$this->api_token_decoded = $this->decode_jwt( $this->api_token );
-			} catch ( Exception $e ) {
-				// If we can't decode the token, try a client credentials grant below.
-				$this->api_token = null;
-			}
-		}
-
-		// Could not decode the stored API token or none was found so try to get one via API.
-		if ( ! $this->api_token_decoded && $this->api_client_creds instanceof WP_Auth0_Api_Client_Credentials ) {
-			$this->api_token         = $this->api_client_creds->call();
-			$this->api_token_decoded = $this->api_client_creds->get_token_decoded();
+		// No stored API token so need to get a new one.
+		if ( ! $api_token && $this->api_client_creds instanceof WP_Auth0_Api_Client_Credentials ) {
+			$api_token = $this->api_client_creds->call();
```

Comment:
is this call aware of the required `$scope`?

Reply:
No, it doesn't need to be. You get the scopes back that the Application is authorized for.